networkit/cpp/distance/DynBFS.cpp

```diff
@@ -42,61 +75,113 @@ void DynBFS::update(GraphEvent e) {
-                if (distances[w] == distances[z]+1) {
-                    if (storePreds) {
-                        previous[w].push_back(z);
+            edgeweight con = infDist;
+            // check whether there are still predecessors for w at level m
+            G->forInNeighborsOf(w, [&](node z) {
+                if (distances[z] != infDist && distances[z] + 1 < con) {
+                    con = distances[z] + 1;
+                }
+            });
+            if (con == m) {
+                npaths[w] = 0;
+                distances[w] = m;
+                visited.push(w);
+                color[w] = BLACK;
+                G->forInNeighborsOf(w, [&](node z) {
+                    // z is a predecessor for w
+                    if (distances[w] == distances[z]+1) npaths[w] += npaths[z];
+                });
+                G->forNeighborsOf(w, [&](node z) {
+                    // w is a predecessor for z
+                    if (distances[w] == infDist || distances[z] >= double(m) + 1.0) {
+                        DEBUG("Pushing ", m + 1, ", max dist = ", maxDistance);
+                        queues[m + 1].push(z);
+                        color[z] = GRAY;
+                        DEBUG("Pushed");
                     }
-                    npaths[w] += npaths[z];
+                });
+            } else {
+                DEBUG("con. value: ", con, "  level: ", m);
+                assert(con > m);				
+                if (distances[w] != infDist) {
+                    npaths[w] = 0;
+                    distances[w] = infDist;
+                    G->forNeighborsOf(w, [&](node z) {
+                        // w was a predecessor for z
+                        if (distances[z] > m && distances[z] < infDist) {
+                            DEBUG("distances[z] = ", distances[z], ", m = ", m);
+                            queues[m + 1].push(z);
+                            color[z] = GRAY;
+                        }
+                    });
                 }
-                //w is a predecessor for z
-                else if (color[z] == WHITE && distances[z] >= distances[w]+1 ) {
-                    color[z] = GRAY;
-                    queues[m+1].push(z);
+                if (con != infDist) {
+                    DEBUG("I'm node ", w, " and I'm putting myself into Q at level ", con,
+                          ". MaxDist = ", maxDistance, ". m = ", m);
+                    if (con > maxDistance) {
+                        DEBUG("CON > MAX DIST. Con = ", con, "maxDist = ", maxDistance);
+                        for (count i = maxDistance; i < con; i++)
+                            queues.emplace_back(std::queue<node>());
+                        maxDistance = con;
+                    }
+                    queues[con].push(w);
                 }
-            });
+            }
         }
-        m = m+1;
     }
-
     // reset colors
-    while(!visited.empty()) {
+    while (!visited.empty()) {
```

Comment:
This change is not needed. Either remove it or format the whole file and remove the `// no-networkit-format` line on top.